internal/traceprof/traceproftest/traceprof_test.go

```diff
@@ -234,3 +236,32 @@ func BenchmarkEndpointsAndHotspots(b *testing.B) {
 		})
 	}
 }
+
+// Test that overriding the resource name for a span & endpoint is reflected in
+// the profile.
+func TestOverrideResourceName(t *testing.T) {
+	tracer.Start(tracer.WithProfilerEndpoints(true))
+	defer tracer.Stop()
+
+	// Running for an arbitrary amount of time is a possible source of
+	// flakiness, but the most we can do is keep trying longer and longer
+	// until what we want shows up in the CPU profile.
+	duration := 100 * time.Millisecond
+	maxDuration := 5 * time.Second
+	for duration <= maxDuration {
+		cp := StartCPUProfile(t)
+		span := tracer.StartSpan("testoverride", tracer.ResourceName("testoverride.old"))
+		span.SetTag(ext.ResourceName, "testoverride.new")
+		stop := make(chan struct{})
+		go cpuHogUntil(stop)
+		time.Sleep(duration)
+		close(stop)
+		span.Finish()
+
+		prof := cp.Stop(t)
+		if prof.LabelDuration(traceprof.TraceEndpoint, "testoverride.new") > 0 {
+			break
+		}
+		duration *= 2
+	}
```

Comment:
🐞 This test passes if I comment out the changes in `ddtrace/tracer/span.go`. I think the `break` should be a `return` and a `t.Fatal(...)` needs to be put after the loop.

Reply:
Ah, great catch, thank you! originally had the test just run forever (avoiding flakiness by just trying longer and longer to get the samples) and relied on the timeout to catch failure. I decided against that but forgot to add a failure. I've pushed an update.

Reply:
No worries. One of my default code review techniques is to break the implementation and see if the included tests start failing. I might have missed this otherwise myself. 😅